packages/flutter_tools/test/general.shard/compute_dev_dependencies_test.dart

```diff
@@ -30,393 +29,110 @@ const String _dartBin = 'bin/cache/dart-sdk/bin/dart';
-          "kind": "root",
-          "dependencies": [
-            "package_a"
-          ],
-          "directDependencies": [
-            "package_a"
-          ]
-        },
-        {
-          "name": "package_a",
-          "kind": "direct",
-          "dependencies": [],
-          "directDependencies": []
-        }
-      ]
-    }''', project: project);
-
-    final Set<String> dependencies = await computeExclusiveDevDependencies(
-      pub(processes),
-      project: project,
-      logger: logger,
-    );
-
-    expect(
-      dependencies,
-      isEmpty,
-      reason: 'No devDependencies: [] specified but still parsed successfully',
-    );
-  });
-
-  test('throws and logs on invalid JSON', () async {
-    final ProcessManager processes = _dartPubDepsReturns('''
-    {
-      "root": "my_app",
-      "packages": [
-        {
-          "name": "my_app",
-          "kind": "root",
-          "dependencies": [
-            "package_a",
-            "package_b"
-          ],
-          "directDependencies": [
-            1
-          ],
-          "devDependencies": []
-        },
-        {
-          "name": "package_a",
-          "kind": "direct",
-          "dependencies": [
-            "package_b"
-          ],
-          "directDependencies": [
-            "package_b"
-          ]
-        },
-        {
-          "name": "package_b",
-          "kind": "transitive",
-          "dependencies": [],
-          "directDependencies": []
-        }
-      ]
-    }''', project: project);
-
-    await expectLater(
-      computeExclusiveDevDependencies(pub(processes), project: project, logger: logger),
-      throwsA(
-        isA<StateError>().having(
-          (StateError e) => e.message,
-          'message',
-          contains('dart pub deps --json had unexpected output'),
+    await validateComputeGraph(
+      <Package>[
+        (
+          name: 'my_app',
+          dependencies: <String>['package_a'],
+          devDependencies: <String>['package_b', 'package_c'],
         ),
-      ),
-    );
-
-    expect(
-      logger.traceText,
-      contains('"root": "my_app"'),
-      reason: 'Stdout should include the JSON blob',
+        (name: 'package_a', dependencies: <String>['package_b'], devDependencies: <String>[]),
+        (name: 'package_b', dependencies: <String>[], devDependencies: <String>[]),
+        (name: 'package_c', dependencies: <String>[], devDependencies: <String>[]),
+      ],
+      <String>['package_c'],
     );
   });
 }
 
-ProcessManager _dartPubDepsReturns(String dartPubDepsOutput, {required FlutterProject project}) {
-  return FakeProcessManager.list(<FakeCommand>[
-    FakeCommand(
-      command: const <String>[_dartBin, 'pub', '--suppress-analytics', 'deps', '--json'],
-      stdout: dartPubDepsOutput,
-      workingDirectory: project.directory.path,
-    ),
-  ]);
+/// A fake [Logger] that throws the [Invocation] for any method call.
+class FakeLogger implements Logger {
```

Comment:
I don't think we should do this (see comment above), but in general this is a better pattern:

```
class FakeLogger extends Fake implements Logger {}
```

It has the behavior you are looking and just dispatches the noral NSM, not requiring an ignore or anything.